smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/ClientConfigProperty.kt

```diff
@@ -237,6 +244,7 @@ object KotlinClientRuntimeConfigProperty {
             NOTE: The caller is responsible for managing the lifetime of the engine when set. The SDK
             client will not close it when the client is closed.
             """.trimIndent()
+            order = -1
```

Comment:
Placing it at `-1` means nothing can be inserted in-between this property and something else at `0`. I'd add a bit of room for growth here (e.g. `-100`).